### PR TITLE
IGNITE-14355 CDC Metrics

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java [hunk 1]
```diff
@@ -111,16 +132,69 @@ public boolean waitForSize(
             timeout);
     }
 
+    /** */
+    public long checkMetrics(CdcMain cdc, int expCnt) throws Exception {
```
Comment (anton-vinogradov) [inline modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java:136]: expCnt never checked

File: modules/core/src/test/java/org/apache/ignite/cdc/CdcSelfTest.java [hunk 2]
```diff
@@ -246,19 +275,34 @@ public void testMultiNodeConsumption() throws Exception {
...
-        addDataFut = runAsync(() -> addData(cache, KEYS_CNT, KEYS_CNT * 2));
+        runAsync(() -> addData(cache, KEYS_CNT, KEYS_CNT * 2)).get(getTestTimeout());
 
-        addDataFut.get(getTestTimeout());
+        // Wait while predicate will become true and state saved on the disk for both cdc.
+        assertTrue(latch.await(getTestTimeout(), MILLISECONDS));
 
-        assertTrue(waitForSize(KEYS_CNT * 2, DEFAULT_CACHE_NAME, UPDATE, getTestTimeout(), cnsmr1, cnsmr2));
+        long evtsCnt1 = checkMetrics(cdc1, -1);
+        long evtsCnt2 = checkMetrics(cdc2, -1);
```
Comment (anton-vinogradov) [inline modules/core/src/test/java/org/apache/ignite/cdc/CdcSelfTest.java:303]:
```suggestion
        long evtsCnt1 = checkMetrics(cdc1, keysCnt[0]);
        long evtsCnt2 = checkMetrics(cdc2, keysCnt[1]);
```
please fix this way other -1s

File: modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java [hunk 3]
```diff
@@ -96,29 +183,87 @@ public void addAndWaitForConsumption(
...
-        return cdcCfg;
+        assertTrue(longMetric.apply(COMMITTED_SEG_OFFSET) >= 0);
+        assertTrue(longMetric.apply(LAST_SEG_CONSUMPTION_TIME) > 0);
+
+        assertTrue(longMetric.apply(LAST_EVT_TIME) > 0);
+
+        for (String m : new String[] {BINARY_META_DIR, MARSHALLER_DIR, CDC_DIR})
+            assertTrue(new File(strMetric.apply(m)).exists());
+
+        if (expCnt != -1)
```
Comment (anton-vinogradov) [inline modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java:253]: we should always check this, see no reason for such ignor.

**Review by anton-vinogradov — APPROVED** [review]

File: modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java (hunk 3 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java:253]: fixed.

File: modules/core/src/test/java/org/apache/ignite/cdc/CdcSelfTest.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/test/java/org/apache/ignite/cdc/CdcSelfTest.java:303]: Yes. Thank you.

**Pull request merged**